x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportStartDataFrameAnalyticsAction.java

```diff
@@ -599,21 +601,24 @@ public void onFailure(Exception e) {
         private final DataFrameAnalyticsAuditor auditor;
         private final MlMemoryTracker memoryTracker;
         private final IndexNameExpressionResolver resolver;
+        private final IndexTemplateConfig inferenceIndexTemplate;
 
         private volatile int maxMachineMemoryPercent;
         private volatile int maxLazyMLNodes;
         private volatile int maxOpenJobs;
         private volatile ClusterState clusterState;
 
         public TaskExecutor(Settings settings, Client client, ClusterService clusterService, DataFrameAnalyticsManager manager,
-                            DataFrameAnalyticsAuditor auditor, MlMemoryTracker memoryTracker, IndexNameExpressionResolver resolver) {
+                            DataFrameAnalyticsAuditor auditor, MlMemoryTracker memoryTracker, IndexNameExpressionResolver resolver,
+                            IndexTemplateConfig inferenceIndexTemplate) {
```

Comment:
I am not sold on this. Since `INFERENCE_TEMPLATE` is public and static, I am wondering if `TransportStartDataFrameAnalyticsAction.TaskExecutor.java` has a static array that it creates of all the templates it cares about.

Reply:
`TaskExecutor` has just 1 template it cares about today. If I use an array I have to iterate that array using call backs in each put request this harms readability and is confusing as there is no requirement for that code today. If in future multiple templates need to be Put the change can be made then.

Using a static reference is fine but it would have to be initialised from the static member in `MlIndexTemplateRegistry` meaning this class would have to 'know' about the template registry as well as `IndexTemplateConfig`.

Reply:
>  meaning this class would have to 'know' about the template registry as well as `IndexTemplateConfig`

Its just a class path. If you would rather extract the template configs to another class, then thats fine too.

The complexity of multiple action listeners is assuaged by `org.elasticsearch.action.support.GroupedActionListener`.

I won't "die on this hill", but I do think adding a parameter here doesn't really make sense.